google-compute-engine/src/main/java/org/jclouds/googlecomputeengine/compute/functions/ResetWindowsPassword.java

```diff
@@ -0,0 +1,241 @@
+            logger.warn("Generating windows key for %s failed. Http Error Code: %d HttpError: %s",
+                  operation.get().targetId(), operation.get().httpErrorStatusCode(),
+                  operation.get().httpErrorMessage());
+         }
+
+         TimeLimiter limiter = new SimpleTimeLimiter();
+         String encryptedPassword = null;
+         try {
+            encryptedPassword = limiter.callWithTimeout(new Callable<String>() {
+               public String call() {
+                  String serialPortContents = "";
+                  Map<String, String> passwordDict = ImmutableMap.of();
+                  while (!serialPortContents.startsWith("{\"ready\":true") || passwordDict.get("encryptedPassword") == null) {
+                     try {
+                        Thread.sleep(30 * 1000);
+                     } catch (InterruptedException e) {
+                        Throwables.propagate(e);
+                     }
+                     serialPortContents = instanceApi.getSerialPortOutput(instance.get().name(), 4).contents();
+                     String[] contentEntries = serialPortContents.split("\n");
+                     passwordDict = new Gson().fromJson(contentEntries[contentEntries.length - 1], Map.class);
+                  }
+                  return passwordDict.get("encryptedPassword");
+               }
+            }, 5, TimeUnit.MINUTES, false); // Notice that timeoutDuration should be less than EXPIRE_DURATION
+         } catch (Exception e) {
+            throw Throwables.propagate(e);
+         }
+
+         return decryptPassword(checkNotNull(encryptedPassword, "encryptedPassword shouldn't be null"), keys);
+
+      } catch (NoSuchAlgorithmException e) {
+         throw Throwables.propagate(e);
+      } catch (InvalidKeySpecException e) {
+         throw Throwables.propagate(e);
+      } catch (NoSuchPaddingException e) {
+         throw Throwables.propagate(e);
+      } catch (InvalidKeyException e) {
+         throw Throwables.propagate(e);
+      } catch (IllegalBlockSizeException e) {
+         throw Throwables.propagate(e);
+      } catch ( BadPaddingException e) {
+         throw Throwables.propagate(e);
+      }
+   }
+
+   /**
+    * Decrypts the given password - the encrypted text is base64-encoded.
+    * As per the GCE docs, assumes it was encrypted with algorithm "RSA/NONE/OAEPPadding", and UTF-8.
+    * @throws NoSuchPaddingException
+    * @throws NoSuchAlgorithmException
+    * @throws InvalidKeyException
+    * @throws BadPaddingException
+    * @throws IllegalBlockSizeException
+    */
+   protected String decryptPassword(String message, KeyPair keys) throws NoSuchAlgorithmException,
+           NoSuchPaddingException, InvalidKeyException, IllegalBlockSizeException, BadPaddingException {
+      // FIXME how to ensure crypto supports OAEPPadding?
+      // FIXME Valentin's code was also passing provider "BC"
+      // Assumes user has configured appropriate crypto guice module.
```

Comment:
It is OK to assume that. We could try/catch the exception here and print a log message suggesting the addition of the bouncycastle driver dependency.